src/main/java/jsr223/perl/utils/PerlVersionGetter.java

```
@@ -25,53 +25,63 @@
+
+    private PerlProcessBuilderFactory factory = PerlSingletonPerlProcessBuilderFactory.getInstance();
 
     /**
      * Retrieves the Perl version.
      *
-     * @return The currently installed version return by the perl command or an empty string
+     * @return The currently installed version return by the perl command or an string that
      * the version could not be determined.
      */
-    public String getPerlVersion(PerlProcessBuilderFactory factory) {
-        if (factory == null) {
-            return "";
-        }
+    public String getPerlVersion() {
+        String result = PERL_VERSION_IF_NOT_INSTALLED; // Default error string for result if version recovery fails
 
-        String result = ""; // Empty string for empty result if version recovery fails
+        PerlCommandCreator perlCommandCreator = new PerlCommandCreator();
+        PerlScriptFileWriter perlScriptFileWriter = new PerlScriptFileWriter();
+        File perlFile = null;
+        try {
+            perlFile = perlScriptFileWriter.forceFileToDisk(PERL_VERSION_COMMAND);
 
-        ProcessBuilder processBuilder = factory.getProcessBuilder(PerlCommandCreator.getPerlCommand(), "-v");
+            String[] perlCommand = perlCommandCreator.createPerlExecutionCommand(perlFile);
 
-        try {
+            ProcessBuilder processBuilder = factory.getProcessBuilder(perlCommand);
             Process process = processBuilder.start();
 
             // Attach stream to std output of process
             StringWriter commandOutput = new StringWriter();
+
             processBuilderUtilities.attachStreamsToProcess(process, commandOutput, null, null);
 
             // Wait for process to exit
             process.waitFor();
 
             // Extract output
             result = commandOutput.toString();
-        } catch (IOException | InterruptedException | IndexOutOfBoundsException e) {
-            log.warn("Failed to retrieve perl version.", e);
+        } catch (Exception e) {
```

Please put a comment explaining why the exception is ignored.